designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceImpl.scala

```diff
@@ -89,21 +97,25 @@ class DeploymentServiceImpl(dispatcher: DeploymentManagerDispatcher,
   // - deployment on engine side - it is longer part, the result will be shown as a notification
   override def deployProcessAsync(processIdWithName: ProcessIdWithName,
                                   savepointPath: Option[String],
-                                  deploymentComment: Option[DeploymentComment])
+                                  comment: Option[String])
                                  (implicit user: LoggedUser, ec: ExecutionContext): Future[Future[Option[ExternalDeploymentId]]] = {
     val actionType = ProcessActionType.Deploy
     checkCanPerformActionAndAddInProgressAction[CanonicalProcess](processIdWithName.id, actionType, d => Some(d.processVersionId), d => Some(d.processingType)).flatMap {
       case (processDetails, actionId, versionOnWhichActionIsDone, buildInfoProcessIngType) =>
-        validateBeforeDeploy(processDetails, actionId).transformWith {
-          case Failure(ex) =>
-            dbioRunner.runInTransaction(actionRepository.removeAction(actionId)).transform(_ => Failure(ex))
-          case Success(validationResult) =>
-            //we notify of deployment finish/fail only if initial validation succeeded
-            val deploymentFuture = runDeploymentActionWithNotifications(actionType, actionId, processIdWithName, versionOnWhichActionIsDone, deploymentComment, buildInfoProcessIngType) {
-              dispatcher.deploymentManagerUnsafe(processDetails.processingType)
-                .deploy(validationResult.processVersion, validationResult.deploymentData, validationResult.resolvedScenario, savepointPath)
+        DeploymentComment.createDeploymentComment(comment, deploymentCommentSettings) match {
```

Comment:
Would be great to flatten this code somehow. One option that I see is to extract a private method for handling the result of `DeploymentComment.createDeploymentComment` and having a for-comprehension on Futures. It requires that this private method return Future as well.

Reply:
Flattened, added for-comprehensions and added a comment with some explanation that this step will be changed.